### PR TITLE
Added Polygon benchmark

### PR DESCRIPTION
**Pull request by OmarWagih1**
Continuing on the effort in benchmarking sympy, i started by benchmarking the polygon class in the geometry folder.

The benchmarks currently work with asv and i'm looking to benchmark other classes in the geometry module once i get some feedback on this benchmark.

#### Release Notes

* benchmarks
  * Added polygon class benchmarking.

### CODE REVIEW

**Comment by sylee957** [comment]
I think that some benchmarks are failing. Did you confirm that you can run the example across some previous versions?

**Comment by OmarWagih1** [comment]
Hello, yes some of them are failing for versions 1.0 - 1.4, but that is because these functions were implemented in version 1.5, starting version 1.5 and up, they should work fine.

I assumed even if we had little history of how they benchmark we would want to benchmark them for the future and start somewhere.

**Comment by OmarWagih1** [comment]
> I think that some benchmarks are failing. Did you confirm that you can run the example across some previous versions?

Hello, yes some of them are failing for versions 1.0 - 1.4, but that is because these functions were implemented in version 1.5, starting version 1.5 and up, they should work fine.

I assumed even if we had little history of how they benchmark we would want to benchmark them for the future and start somewhere.

**Comment by sylee957** [comment]
We should look for a way to automatically disable some benchmarks if they are not supported in old versions

**Comment by OmarWagih1** [comment]
> We should look for a way to automatically disable some benchmarks if they are not supported in old versions

The only way i could find in asv to skip tests is to raise a NotImplementedError in the setup class, this is skipped in functions intentionally by ASV (Could refer to this issue https://github.com/airspeed-velocity/asv/issues/638),

so the current way to skip benchmarks is by making them all in separate classes, not ideal coding format as i assume, but the only solution I could find. Benchmarks are now skipped (output is n/a) if the function is not supported.

File: benchmarks/polygon.py [hunk 1]
```diff
@@ -4,69 +4,125 @@
...
-	    [self.listOfPolygons[w].centroid for w in range(10)]
+	    [listOfPolygons[w].centroid for w in range(10)]
 
-	def time_polygon_second_moment(self):
+class PolygonSecondMoment:
+	def setup(self):
+		try:
+			listOfPolygons[0].second_moment_of_area()
+		except AttributeError:
+			raise NotImplementedError
```
Comment (smitgajjar) [inline benchmarks/polygon.py]: Make sure there are 4 spaces per indentation level, according to PEP8

**Comment by sylee957** [comment]
What about grouping the benchmarks into things that were working persistently across all versions, and things that only works after 1.5?

And I think that you should better use sympy.__version__ than catching the attribute errors, because someone can introduce some bugs for the methods later.

**Comment by OmarWagih1** [comment]
> What about grouping the benchmarks into things that were working persistently across all versions, and things that only works after 1.5?
> 
> And I think that you should better use sympy.**version** than catching the attribute errors, because someone can introduce some bugs for the methods later.

They don't all appear in version 1.5, of the ones that are not in 1.0, there is one that is developed in 1.3 and others that are developed in 1.5, i grouped the attributes together under an umbrella class and the other functions were left as is, I think it would be more descriptive this way.

And yes, that is a very good note, thank you! i used the sympy.__version__ to compare.

**Pull request merged**